src/Rules/SyntaxErrorInDibiPreparedStatementMethodRule.php

```diff
@@ -143,7 +144,7 @@ private function checkErrors(CallLike $callLike, Scope $scope, MethodReflection
         }
 
         $placeholders = [];
-        preg_match_all('#%(ex|in|i|and|or|s|t|d|~?like~?|n|lmt|ofs)#', $queryParameters[0], $placeholders, \PREG_SET_ORDER);
+        preg_match_all('#%(sN|bin|by|lmt|b|iN|f|dt|sql|ex|in|i|l|m|and|or|s|t|d|~?like~?|n|ofs|N)#', $queryParameters[0], $placeholders, \PREG_SET_ORDER);
         $placeholderCount = \count($placeholders);
         $parameterCount = \count($queryParameters) - 1;
 
```

Comment:
line 151 checks for `INSERT`... what about `UPDATE` , `DELETE`, `ALTER TABLE` etc. do all other query types support these placeholders?

or what it make more sense to whitelist the types which actually support the placeholders instead of doing blacklisting?

Reply:
here I define the list of all dibi placeholders and I count how many of them were found.

The number should match number of parameters. Regardless whether its `insert, update, alter table` statelement. The two numbers should match else we know for sure that the placeholder validation got to fail.

Reply:
jep, I am talking about line 151 (which was not changed with this PR, but jumped into my eye while reviewing)

Reply:
thanks